sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionEvaluatesToConstant.cs

```diff
@@ -132,6 +145,16 @@ private static void CheckForRedundantConditions(CSharpExplodedGraph explodedGrap
                         .Except(conditionTrue)
                         .Where(c => !IsInsideCatchOrFinallyBlock(c))
                         .Select(node => GetDiagnostics(node, false)))
+                    .Union(isNull
+                        .Except(isUnknown)
+                        .Except(isNotNull)
+                        .Where(c => !IsInsideCatchOrFinallyBlock(c))
+                        .Select(node => Diagnostic.Create(s2583Null, node.GetLocation(), messageArgs: "null")))
```

Comment:
For this case, I'd raise the `S2589` version of the check, as the right hand side of the code will always be executed, so the condition is useless. This falls in the code smell rule case